### ISSUE TITLE
Upgrading 0.2.19 to 0.2.15 causes migrations to be ORDER BY "id" instead of timestamp

### ISSUE THREAD

**Issue by thatcort** [typeorm/typeorm#4941]
**Issue type:**

[ ] question
[x] bug report
[ ] feature request
[ ] documentation issue

**Database system/driver:**

[ ] `cordova`
[ ] `mongodb`
[ ] `mssql`
[ ] `mysql` / `mariadb`
[ ] `oracle`
[x] `postgres`
[ ] `cockroachdb`
[ ] `sqlite`
[ ] `sqljs`
[ ] `react-native`
[ ] `expo`

**TypeORM version:**

[x] `latest`
[ ] `@next`
[ ] `0.x.x` (or put your version here)

**Steps to reproduce or a small repository showing the problem:**

I just upgraded to TypeORM 0.2.20 and it now fails to start. This is due to the MigrationExecutor expecting an ID column in the migration table when it runs `loadExecutedMigrations`:

`SELECT * FROM "migrations" "migrations" ORDER BY id DESC`

In my DB there is only a `timestamp` and a `name` column.

The `ORDER BY id` clause was added 3 days ago in fix #4702.

I'm not sure when the id column was added to the table. My database was created with TypeORM 0.1.x and I recently upgraded to 0.2.19 and everything was working fine. The 0.2.20 update breaks it.

**Comment by pleerock** [comment]
@RecuencoJones @rustamwin can you please check this out?

**Comment by RecuencoJones** [comment]
Hi @thatcort, @pleerock, 

The only change I see that could be causing this is the line: https://github.com/typeorm/typeorm/blob/master/src/migration/MigrationExecutor.ts#L320, which attempts to sort all migrations by ID as agreed on https://github.com/typeorm/typeorm/issues/4697

However, shouldn't all migrations be assigned an ID by default when running this? https://github.com/typeorm/typeorm/blob/e42091e1506148ba761737d7977331b5c1d60307/src/migration/MigrationExecutor.ts#L389

**Comment by thatcort** [comment]
My migration table does not have an ID column. They were created with TypeORM v0.1.x and upgraded recently to 0.2.19.

Looking back through the Blame history on MIgrationExecutor.ts, it seems the ID column was added in March 2018 as part of the fix for #1315: https://github.com/typeorm/typeorm/commit/327f419a9d0b6158c78bcd56145a145ac4025148

Reading through it, I don't see how existing migration tables are migrated to the new schema. Can you point me to where this is covered?

**Comment by emlynmac** [comment]
I too, am running into this problem - was just about to raise another issue.
I just went from 0.2.19 to 0.2.20 and the issue has emerged.
My db was originally generated in the 0.1 version but has been fine until now.

SELECT * FROM "migrations" "migrations" ORDER BY id DESC

**Comment by imnotjames** [comment]
Is this still an issue in the latest release?

**Comment by thatcort** [comment]
Yes, it will bite you when you upgrade from a previous release to 0.2.20 or later. You can work around it by manually adding the ID column to the migration table of the DB.

**Comment by imnotjames** [comment]
Interesting.. I've seen a number of things where the migration table has caused issues.
